job-server/src/main/scala/spark/jobserver/AkkaClusterSupervisorActor.scala

```diff
@@ -124,20 +124,31 @@ class AkkaClusterSupervisorActor(daoActor: ActorRef, dataManagerActor: ActorRef)
           (contextFromDAO, callbacks) match {
             case (Some(JobDAOActor.ContextResponse(Some(contextInfo))),
                 Some((successCallback, failureCallback))) =>
-              val contextConfig = ConfigFactory.parseString(contextInfo.config)
-              val isAdhoc = contextConfig.getBoolean("is-adhoc")
-              initContext(contextConfig, actorName,
-                     actorRef, contextInitTimeout)(isAdhoc, successCallback, failureCallback)
+              initContext(contextInfo, actorName, actorRef, false)(successCallback, failureCallback)
             case (Some(JobDAOActor.ContextResponse(None)), _) =>
               logger.error(
                  s"No such contextId ${contextId} was found in DB. Cannot initialize actor ${actorName}")
               actorRef ! PoisonPill
             case (Some(JobDAOActor.ContextResponse(Some(contextInfo))), None) =>
-              val exception = NoCallbackFoundException(contextInfo.id, actorRef.path.toSerializationFormat)
-              logger.error(exception.getMessage, exception)
-              actorRef ! PoisonPill // Since no watch was added, Terminated will not be fired
-              daoActor ! JobDAOActor.SaveContextInfo(contextInfo.copy(
-                  state = ContextStatus.Error, endTime = Some(DateTime.now()), error = Some(exception)))
+              isSuperviseModeEnabled(config, ConfigFactory.parseString(contextInfo.config)) match {
+                case isRestartScenario @ true =>
+                  logger.info(
```

Comment:
Is it that restart request is received right at this moment (which makes it seem like the context is about to be restarted), or do you mean that supervise mode is true, therefore restarts are enabled (which is what I read out of the code here)

Reply:
Your previous comment is correct.

Just to add here, If context is started with supervise mode then on unexpectedly terminated, it will be terminated and will be restarted by Spark. As the Context JVM starts up, it will connect with the SJS HTTP JVM and will land in ActorIdentity. Here we just check that if this context is eligible for restart or not. If yes, we let it thorugh (which is almost always) and stop it other wise.